src/tclint/format.py

```diff
@@ -138,7 +168,9 @@ def format_partial(self, script: str, parser: Parser) -> str:
         tree = parser.parse(script)
         self.script = script.split("\n")
 
-        formatted = "\n".join(self._indent(self.format_script_contents(tree), indent))
+        lines = self._indent(self.format_script_contents(tree), indent)
+        lines = self.reindent(lines)
+        formatted = "\n".join(lines)
```

Comment:
This approach looks good, although I think there's one problematic edge case here - if the start of the partial fragment is indented with any tab characters, then the `self.reindent()` call will behave improperly since the tabs will be propagated to each line but reindent expects spaces.